docs/blog/2019-04-04-how-to-build-a-blog-with-wordpress-and-gatsby-part-2/index.md

```diff
@@ -0,0 +1,408 @@
+}
+```
+
+### Making sure plugins are working
+
+If the Gatsby site is currently running, you need to stop it and restart it so it pulls in the new content from WordPress. It's important to know that while you can choose what information goes on what pages after your app is built, it will only pull content when it is initially run so changes in the source require a rebuild.
+
+Once you've restarted your server, you can visit [http://localhost:8000/\_\_\_graphql](http://localhost:8000/___graphql) to use the "graphical" playground. Here, you can use GraphQL to query your data for testing purposes. You should create opening and closing curly braces and then you can use shift+space (or ctrl+space on windows) to get suggestions. Once you have the data you want, you will be able to paste the query into your components, pages, and templates so you can use the information available. Here's what my query looks like for now:
+
+![GraphQL query tests](images/gatsby2.png)
+
+You may notice that there are several drilldowns inside of the `acf` field. This is saying "hey, look for the ACF field called feat_img and get the local, optimized versions of these images so you can use them". Gatsby also provides fragments which means inside of your application you could just put `...GatsbyImageSharpSizes` instead of drilling down so far and Gatsby will know what to do with it.
+
+Since you are seeing stuff on the right-hand side, it means that you are getting your data from WordPress, which is awesome! Now you need to tell Gatsby what to do with the data, so let's talk about the `gatsby-node.js` file a bit.
+
+## Creating pages in gatsby-node.js
+
+As I briefly mentioned earlier, the `gatsby-node.js` file is there so you can build pages programatically from data. There are two pieces to make this work: the logic in `gatsby-node.js` and a template file to render the data. Let's start by creating a simple template with no dynamic data just to make sure your logic is working properly.
+
+### Creating templates in Gatsby.js
+
+If you look inside the `src/` folder, you can see directories for components, images, and pages. You need to add one that will house your templates and then add a template for your blog posts. The following code will do this for you:
+
+```shell
+mkdir ./src/templates && touch ./src/templates/BlogPost.js
+```
+
+Unless there is some special functionality needed for blog posts, you can just [use a stateless functional component](https://www.iamtimsmith.com/blog/class-components-vs-stateless-functional-components/). I won't go over components in this post, but if you need a refresher, you can [read up on components here](https://www.iamtimsmith.com/blog/how-to-create-a-component/).
+
+Below is the code I'm using for the template starter. This is just to make sure things are working after you finish setting up your `gatsby-node.js` logic. Once you know the page is being created, you will update the template to display correctly.
+
+```javascript
```

Comment:
```suggestion
```javascript:title=src/templates/BlogPost.js
```

Also would be interesting if we add a directive, e.g. something like

```
/* title=src/templates/BlogPost.js */
```

as the first line of a code block. But--outside the scope of this PR!

Reply:
@DSchau Is that possible currently? I'd be happy to do that since I'm not thrilled about the commented file name look. If not, it would be an awesome addition!

Reply:
Not possible currently! Future work--if you're interested in making this change, it'd be adding some functionality to gatsby-remark-prismjs!

Reply:
@DSchau Sure, I'd be interested in looking into that. I assume that will be a separate issue though. Should I just work on it and create the issue with the PR when I have a solution?

Reply:
Definitely!